src/package.cc

```diff
@@ -60,6 +60,99 @@ Package::Package(std::string name, double fill_min, double fill_max,
         throw ValueError("can't create a new package with name 'unpackaged'");
       }
   }
+    if (strategy != "first" && strategy != "equal") {
+      throw ValueError("Invalid strategy for package: " + strategy_);
+    }
+}
+
+TransportUnit::Ptr TransportUnit::unrestricted_ = NULL;
+
+TransportUnit::Ptr TransportUnit::Create(std::string name, int fill_min, int fill_max, std::string strategy) {
+  if (fill_min < 0 || fill_max < 0) {
+    throw ValueError("fill_min and fill_max must be non-negative");
+  }
+  else if (fill_min > fill_max) {
+    throw ValueError("fill_min must be less than or equal to fill_max");
+  }
+  Ptr t(new TransportUnit(name, fill_min, fill_max, strategy));
+  return t;
+}
+
+// singleton pattern: 
+// if the static member is not yet set, create a new object
+// otherwise return the object that already exists
+TransportUnit::Ptr& TransportUnit::unrestricted() {
+  if (!unrestricted_) {
+    unrestricted_ = Ptr(new TransportUnit(unrestricted_name_));
+  }
+  return unrestricted_;
+}
+
+int TransportUnit::GetTransportUnitFill(int qty) {
+  if (qty < fill_min_) {
+    // less than one TransportUnit of material available
+    return 0;
+  }
+
+  if (strategy_ == "first") {
+    return fill_max_;
+  } else if (strategy_ == "equal" || strategy_ == "hybrid") {
```

Comment:
Since the `"hybrid"` method may have unequal filing, it seems strange to return a single number...

Reply:
Yeah in that case it's more like next fill in that case